src/Microsoft.Health.Dicom.Core/Features/ChangeFeed/ChangeFeedService.cs

```diff
@@ -3,86 +3,77 @@
 // Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
 // -------------------------------------------------------------------------------------------------
 
+using System;
 using System.Collections.Generic;
-using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using EnsureThat;
-using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Dicom.Core.Configs;
 using Microsoft.Health.Dicom.Core.Features.Common;
 using Microsoft.Health.Dicom.Core.Features.Model;
+using Microsoft.Health.Dicom.Core.Models;
 
 namespace Microsoft.Health.Dicom.Core.Features.ChangeFeed;
 
 public class ChangeFeedService : IChangeFeedService
 {
-    private const int MaxLimit = 100;
     private readonly IChangeFeedStore _changeFeedStore;
     private readonly IMetadataStore _metadataStore;
+    private readonly RetrieveConfiguration _options;
 
-    public ChangeFeedService(IChangeFeedStore changeFeedStore, IMetadataStore metadataStore)
+    public ChangeFeedService(
+        IChangeFeedStore changeFeedStore,
+        IMetadataStore metadataStore,
+        IOptions<RetrieveConfiguration> options)
     {
-        EnsureArg.IsNotNull(changeFeedStore, nameof(changeFeedStore));
-        EnsureArg.IsNotNull(metadataStore, nameof(metadataStore));
-
-        _changeFeedStore = changeFeedStore;
-        _metadataStore = metadataStore;
+        _changeFeedStore = EnsureArg.IsNotNull(changeFeedStore, nameof(changeFeedStore));
+        _metadataStore = EnsureArg.IsNotNull(metadataStore, nameof(metadataStore));
+        _options = EnsureArg.IsNotNull(options?.Value, nameof(metadataStore));
     }
 
-    public async Task<IReadOnlyCollection<ChangeFeedEntry>> GetChangeFeedAsync(long offset, int limit, bool includeMetadata, CancellationToken cancellationToken)
+    public async Task<IReadOnlyList<ChangeFeedEntry>> GetChangeFeedAsync(TimeRange range, long offset, int limit, ChangeFeedOrder order, bool includeMetadata, CancellationToken cancellationToken = default)
```

Comment:
Are we logging the offset somewhere? Hopefully this helps to investigate SQL perf issues?

Reply:
We do log that information in the controller